src/Map.ts

```diff
@@ -1190,4 +1208,209 @@ export class Map extends maplibregl.Map {
+    else if (data && !geojsonValidation.valid(data)) {
+      throw new Error(
+        "Polyline data was provided as an object but object is not of a valid GeoJSON format",
+      );
+    }
+
+    return this.addGeoJSONPolyline({
+      ...options,
+      data,
+    });
+  }
+
+  /**
+   * Add a polyline witgh optional outline from a GeoJSON object
+   * @param data
+   * @param options
+   * @returns
+   */
+  private addGeoJSONPolyline(
+    // this Feature collection is expected to contain on LineStrings and MultilLinestrings
+    options: PolylineLayerOptions,
+  ): {
+    polylineLayerId: string;
+    polylineOutlineLayerId: string | null;
+    polylineSourceId: string;
+  } {
+    if (options.layerId && this.getLayer(options.layerId)) {
+      throw new Error(
+        `A layer already exists with the layer id: ${options.layerId}`,
+      );
+    }
+
+    const sourceId = options.sourceId ?? generateRandomSourceName();
+    const layerId = options.layerId ?? generateRandomLayerName();
+
+    const retunedInfo = {
+      polylineLayerId: layerId,
+      polylineOutlineLayerId: "",
+      polylineSourceId: sourceId,
+    };
+
+    // A new source is added if the map does not have this sourceId and the data is provided
+    if (options.data && !this.getSource(sourceId)) {
+      // Adding the source
+      this.addSource(sourceId, {
+        type: "geojson",
+        data: options.data,
+      });
+    }
+
+    const lineWidth = options?.lineWidth ?? 3;
+    const lineColor = options?.lineColor ?? getRandomColor();
+    const lineOpacity = options?.lineOpacity ?? 1;
+    const outlineWidth = options?.outlineWidth ?? 1;
+    const outlineColor = options?.outlineColor ?? "#FFFFFF";
+    const outlineOpacity = options?.outlineOpacity ?? 1;
+
+    // We want to create an outline for this line layer
+    if (options?.outline === true) {
+      const outlineLayerId = `${layerId}_outline`;
+      retunedInfo.polylineOutlineLayerId = outlineLayerId;
+
+      this.addLayer(
+        {
+          id: outlineLayerId,
+          type: "line",
+          source: sourceId,
+          layout: {
+            "line-join": "round",
+            "line-cap": "round",
+          },
+          minzoom: options?.minzoom ?? 0,
+          maxzoom: options?.maxzoom ?? 22,
+          paint: {
+            "line-opacity":
+              typeof outlineOpacity === "number"
+                ? outlineOpacity
+                : lineOpacityOptionsToLineLayerPaintSpec(outlineOpacity),
+            "line-color":
+              typeof outlineColor === "string"
+                ? outlineColor
+                : lineColorOptionsToLineLayerPaintSpec(outlineColor),
+            "line-width": computeRampedOutlineWidth(lineWidth, outlineWidth),
+          },
+        },
+        options?.beforeId,
+      );
+    }
+
+    this.addLayer(
+      {
+        id: layerId,
+        type: "line",
+        source: sourceId,
+        layout: {
```

Comment:
Same case as above comment.

Reply:
Yes, I've just added more options. But the point of the polylineHelper is not to cover 100% of what is feasible with the low level styling. Instead it's about providing an easy way to use the most common styling options.